xml/src/main/java/org/ehcache/xml/CoreCacheConfigurationParser.java

```diff
@@ -72,29 +83,47 @@
     return expiry;
   }
 
+  static <T> T getInstanceOfName(String name, ClassLoader classLoader, Class<T> type) throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+    if (name == null) {
+      return null;
+    }
+    Class<?> klazz = getClassForName(name, classLoader);
+    return klazz.asSubclass(type).newInstance();
+  }
+
   public CacheType unparseConfiguration(CacheConfiguration<?, ?> cacheConfiguration, CacheType cacheType) {
     ExpiryPolicy<?, ?> expiryPolicy = cacheConfiguration.getExpiryPolicy();
     if (expiryPolicy != null) {
+      Duration expiry = expiryPolicy.getExpiryForCreation(null, null);
       ExpiryType expiryType = new ExpiryType();
-      if (expiryPolicy == ExpiryPolicy.NO_EXPIRY) {
+      if (expiryPolicy.equals(ExpiryPolicy.NO_EXPIRY)) {
         expiryType.withNone(new ExpiryType.None());
-      } else if (expiryPolicy instanceof ExpiryPolicyBuilder.TimeToLiveExpiryPolicy) {
-        Duration ttl = expiryPolicy.getExpiryForCreation(null, null);
-        expiryType.withTtl(new TimeType().withValue(BigInteger.valueOf(ttl.getSeconds())).withUnit(TimeUnit.SECONDS));
-      } else if (expiryPolicy instanceof ExpiryPolicyBuilder.TimeToIdleExpiryPolicy) {
-        Duration tti = expiryPolicy.getExpiryForCreation(null, null);
-        expiryType.withTti(new TimeType().withValue(BigInteger.valueOf(tti.getSeconds())).withUnit(TimeUnit.SECONDS));
+      } else if (expiryPolicy.equals(ExpiryPolicyBuilder.timeToLiveExpiration(expiry))) {
+        expiryType.withTtl(convertToTimeType(expiry));
+      } else if (expiryPolicy.equals(ExpiryPolicyBuilder.timeToIdleExpiration(expiry))) {
+        expiryType.withTti(convertToTimeType(expiry));
       } else {
-        expiryType.setClazz(expiryPolicy.getClass().getName());
+        throw new XmlConfigurationException("XML translation of custom expiry policy is not supported");
       }
       cacheType.withExpiry(expiryType);
     }
 
     EvictionAdvisor<?, ?> evictionAdvisor = cacheConfiguration.getEvictionAdvisor();
     if (evictionAdvisor != null) {
-      cacheType.withEvictionAdvisor(evictionAdvisor.getClass().getName());
+      throw new XmlConfigurationException("XML translation of eviction advisor is not supported");
     }
 
     return cacheType;
   }
+
+  private static TimeType convertToTimeType(Duration duration) {
+    return Stream.of(java.util.concurrent.TimeUnit.values())
```

Comment:
This should really be a stream of the jaxb mapped time units that gets mapped to a stream of `j.u.c.TimeUnit`, right?

Reply:
Why? We need the XML time unit only at the end before the final transformation. Until then we're mostly using the `j.u.c.TimeUnit` directly. That's why I started the source from it directly.